integration/kube_integration_test.go

```diff
@@ -593,6 +595,196 @@ func testKubePortForward(t *testing.T, suite *KubeSuite) {
+	_, impersonatingProxyClientConfig, err := kube.ProxyClient(kube.ProxyConfig{
+		T:             teleport,
+		Username:      username,
+		KubeGroups:    kubeGroups,
+		Impersonation: &rest.ImpersonationConfig{UserName: "bob", Groups: []string{kube.TestImpersonationGroup}},
+	})
+	require.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		builder func(*rest.Config, kubePortForwardArgs) (*kubePortForwarder, error)
+	}{
+		{
+			name:    "SPDY",
+			builder: newPortForwarder,
+		},
+		{
+			name:    "SPDY over Websocket",
+			builder: newPortForwarderSPDYOverWebsocket,
+		},
+	}
+
+	// // Setup random generator for pod name uniqueness.
+	// rnd := rand.New(rand.NewSource(time.Now().Unix()))
+	// rndBuf := make([]byte, 4)
+
+	for _, tt := range tests {
+		t.Run(tt.name,
+			func(t *testing.T) {
+				// // Create a new pod.
+				/// // Avoid interfering with singleton pod in test suite.
+				// rnd.Read(rndBuf)
+				// tmpPodName := fmt.Sprintf("%v-%x", testPod, rndBuf)
+				// p := newPod(testNamespace, tmpPodName)
+				// _, err := suite.CoreV1().Pods(testNamespace).Create(context.Background(), p, metav1.CreateOptions{})
+				// require.NoError(t, err)
+				// t.Cleanup(func() {
+				// 	// Later logic will also attempt to delete the pod.
+				// 	// Fine if this cleanup delete has nothing to delete.
+				// 	_ = suite.CoreV1().Pods(testNamespace).Delete(context.Background(), tmpPodName, metav1.DeleteOptions{})
+				// })
+
+				// // Wait for pod to be running.
+				// require.Eventually(t, func() bool {
+				// 	pod, err := suite.CoreV1().Pods(testNamespace).Get(context.Background(), testPod, metav1.GetOptions{})
+				// 	if err != nil {
+				// 		return false
+				// 	}
+				// 	if pod.Status.Phase != v1.PodRunning {
+				// 		return false
+				// 	}
+				// 	if len(pod.Status.ContainerStatuses) == 0 {
+				// 		return false
+				// 	}
+				// 	for _, containerStatus := range pod.Status.ContainerStatuses {
+				// 		if containerStatus.Name == nginx {
+				// 			return containerStatus.Ready
+				// 		}
+				// 	}
+				// 	return false
+				// }, 60*time.Second, time.Millisecond*500)
+
+				t.Cleanup(func() {
+					// Test pod would have been deleted if test is successful.
+					// Setup next test with test pod as needed.
+					pod := newPod(testNamespace, testPod)
+					_, err = suite.CoreV1().Pods(testNamespace).Create(context.Background(), pod, metav1.CreateOptions{})
+					if err != nil {
+						require.True(t, kubeerrors.IsAlreadyExists(err), "Failed to create test pod: %v", err)
+					}
+					// Wait for pod to be running.
+					require.Eventually(t, func() bool {
+						rsp, err := suite.CoreV1().Pods(testNamespace).Get(context.Background(), testPod, metav1.GetOptions{})
+						if err != nil {
+							return false
+						}
+						return rsp.Status.Phase == v1.PodRunning
+					}, 60*time.Second, time.Millisecond*500)
+				})
+
+				// Forward local port to container port.
+				listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
for future reference, I think we don't need this.
We can pass port :0 to the portforward and then parse the first line written to stdout (portforward allows custom stdout io.Writer) 
```
$ kubectl port-forward deployment/test-web 0:80
Forwarding from 127.0.0.1:43123 -> 80
```